tests/backup/backup_kubevirt_test.go

```diff
@@ -0,0 +1,382 @@
+					err := CreateBackupWithValidation(ctx, backupName, SourceClusterName, backupLocationName, backupLocationUID, []*scheduler.Context{appCtx}, labelSelectors, orgID, sourceClusterUid, "", "", "", "")
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupName))
+				}(backupName, appCtx)
+			}
+			wg.Wait()
+		})
+
+		Step("Restoring all individual backups", func() {
+			log.InfoD("Restoring all individual backups")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			var wg sync.WaitGroup
+			restoreNames = make([]string, 0)
+			for i, appCtx := range scheduledAppContexts {
+				restoreName := fmt.Sprintf("%s-%s-%v", "auto-restore", appCtx.ScheduleOptions.Namespace, RandomString(6))
+				restoreNames = append(restoreNames, restoreName)
+				restoreNameToAppContextMap[restoreName] = appCtx
+				wg.Add(1)
+				go func(restoreName string, appCtx *scheduler.Context, i int) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					log.InfoD("Restoring [%s] namespace from the [%s] backup", appCtx.ScheduleOptions.Namespace, backupNames[i])
+					err = CreateRestore(restoreName, backupNames[i], make(map[string]string), destinationClusterName, orgID, ctx, make(map[string]string))
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of restore [%s]", restoreName))
+				}(restoreName, appCtx, i)
+			}
+			wg.Wait()
+		})
+
+		Step("Validating restores of individual backups", func() {
+			log.InfoD("Validating restores of individual backups")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			defer func() {
+				log.InfoD("Switching cluster context back to source cluster")
+				err = SetSourceKubeConfig()
+				log.FailOnError(err, "Switching context to source cluster failed")
+			}()
+			err = SetDestinationKubeConfig()
+			log.FailOnError(err, "Switching context to destination cluster failed")
+			var wg sync.WaitGroup
+			for _, restoreName := range restoreNames {
+				wg.Add(1)
+				go func(restoreName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					log.InfoD("Validating restore [%s]", restoreName)
+					expectedRestoredAppContext, err := CloneAppContextAndTransformWithMappings(restoreNameToAppContextMap[restoreName], make(map[string]string), make(map[string]string), true)
+					log.FailOnError(err, "Failed TransformAppContextWithMappings")
+					err = ValidateRestore(ctx, restoreName, orgID, []*scheduler.Context{expectedRestoredAppContext}, make([]string, 0))
+					log.FailOnError(err, "Failed while validating restore - "+restoreName)
+				}(restoreName)
+
+			}
+			wg.Wait()
+		})
+
+		Step("Take backup of all namespaces with VMs in Running and Stopped state", func() {
+			log.InfoD("Take backup of all namespaces with VMs in Running and Stopped state")
+			ctx, err := backup.GetAdminCtxFromSecret()
```

Comment:
Can you move this into a function an keep call it in all the steps?

Reply:
Which function? You mean `GetAdminCtxFromSecret` ?

Reply:
Can we move the entire code of the step which takes all namespace backup into a single function and keep calling.?

Reply:
Got it. Actually, I thought about the same. But then I realised that if I make it as function I would have to pass a lot of inputs and also return a namespace mapping map which is used in the restore step.
I would have to pass every parameter which we are passing to the CreateBackup.

Reply:
Why no have a simple local function? Whatever is changing you can either pass it as an parameter or just have one parent parameter which can be say 1,2,3,4 which can have a if to do the needed settings?

Reply:
The function will look like this. This will take in 9 different params. Ideally no function should accept so many arguments (like we have CreateBackup today)

```
func TakeBackupOfNamespaces(scheduledAppContexts []*scheduler.Context, namespaceMappingMixed map[string]string, backupNames []string,
	sourceClusterUid, backupWithVMMixed, SourceClusterName, backupLocationName, backupLocationUID, namespaceSuffix string) {
	ctx, err := backup.GetAdminCtxFromSecret()
	log.FailOnError(err, "Fetching px-central-admin ctx")
	var namespaces []string
	for _, appCtx := range scheduledAppContexts {
		namespaces = append(namespaces, appCtx.ScheduleOptions.Namespace)
		namespaceMappingMixed[appCtx.ScheduleOptions.Namespace] = appCtx.ScheduleOptions.Namespace + "-" + namespaceSuffix
	}
	backupNames = append(backupNames, backupWithVMMixed)
	log.InfoD("creating backup [%s] in cluster [%s] (%s), organization [%s], of namespace [%v], in backup location [%s]", backupWithVMMixed, SourceClusterName, sourceClusterUid, orgID, namespaces, backupLocationName)
	err = CreateBackupWithValidation(ctx, backupWithVMMixed, SourceClusterName, backupLocationName, backupLocationUID, scheduledAppContexts,
		nil, orgID, sourceClusterUid, "", "", "", "")
	dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of backup [%s]", backupWithVMMixed))
}
```

This is the reason why I avoided it.

Reply:
Okay lets go ahead.